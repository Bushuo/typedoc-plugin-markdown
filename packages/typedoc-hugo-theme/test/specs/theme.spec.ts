import * as fs from 'fs';

import { TestApp } from 'typedoc-plugin-markdown/test/test-app';
import { PageEvent } from 'typedoc/dist/lib/output/events';

let testApp: TestApp;

describe(`Theme:`, () => {
  beforeAll(async () => {
    testApp = new TestApp(['theme.ts']);
    await testApp.bootstrap({
      theme: './dist',
      entryDocument: '_index.md',
      indexTitle: 'Test API',
    });
  });
  test(`should getUrls'`, async () => {
    const urlMappings = testApp.theme.getUrls(testApp.project);
    expect(TestApp.getExpectedUrls(urlMappings)).toMatchSnapshot();
  });
  describe(`(YAML)`, () => {
    test(`should prepend YAML block to start of page`, async () => {
      const page = {
        project: { url: 'modules.md' },
        model: { name: 'SomeClass', kindString: 'Class' },
        url: 'classes/someclass.md',
        contents: 'CONTENTS',
      } as PageEvent;
      (testApp.theme as any).onHugoPageEnd(page);
      expect(page.contents).toMatchSnapshot();
    });

    test(`should set entry page without a readme`, async () => {
      const page = {
        project: { url: '_index.md' },
        url: '_index.md',
        contents: 'CONTENTS',
      } as PageEvent;
      (testApp.theme as any).onHugoPageEnd(page);
      expect(page.contents).toMatchSnapshot();
    });

    test(`should set entry page with a readme`, async () => {
      const page = {
        project: { url: 'modules.md' },
        url: '_index.md',
        contents: 'CONTENTS',
      } as PageEvent;
      (testApp.theme as any).onHugoPageEnd(page);
      expect(page.contents).toMatchSnapshot();
    });

    test(`should set modules page`, async () => {
      const page = {
        project: { url: 'modules.md' },
        url: 'modules.md',
        contents: 'CONTENTS',
      } as PageEvent;
      (testApp.theme as any).onHugoPageEnd(page);
      expect(page.contents).toMatchSnapshot();
    });
  });

  describe(`(branch bundles)`, () => {
    test(`should write interfaces _index.md`, async () => {
      const interfacesFile = fs.readFileSync(
        testApp.tmpobj.name + '/interfaces/_index.md',
      );
      expect(interfacesFile.toString()).toMatchSnapshot();
    });
    test(`should write classes _index.md`, async () => {
      const interfacesFile = fs.readFileSync(
        testApp.tmpobj.name + '/classes/_index.md',
      );
      expect(interfacesFile.toString()).toMatchSnapshot();
    });
  });
});
