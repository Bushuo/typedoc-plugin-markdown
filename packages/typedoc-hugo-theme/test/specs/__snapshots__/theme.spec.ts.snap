// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: (YAML) should prepend YAML block to start of page 1`] = `
"---
title: \\"Class: SomeClass\\"
linkTitle: \\"SomeClass\\"
slug: \\"someclass\\"
---

CONTENTS"
`;

exports[`Theme: (YAML) should set entry page with a readme 1`] = `
"---
title: \\"Readme\\"
linkTitle: \\"Readme\\"
slug: \\"_index\\"
---

CONTENTS"
`;

exports[`Theme: (YAML) should set entry page without a readme 1`] = `
"---
title: \\"Test API\\"
linkTitle: \\"Test API\\"
slug: \\"_index\\"
---

CONTENTS"
`;

exports[`Theme: (YAML) should set modules page 1`] = `
"---
title: \\"Test API\\"
linkTitle: \\"Test API\\"
slug: \\"modules\\"
---

CONTENTS"
`;

exports[`Theme: (branch bundles) should write classes _index.md 1`] = `
"---
title: \\"Classes\\"
---"
`;

exports[`Theme: (branch bundles) should write interfaces _index.md 1`] = `
"---
title: \\"Interfaces\\"
---"
`;

exports[`Theme: should getUrls' 1`] = `
Array [
  "modules.md",
  "modules.md#declarationitema",
  "modules.md#declarationitemb",
  "modules.md#functionitema",
  "modules.md#functionitemb",
  "_index.md",
  "modules.md#declarationitema",
  "modules.md#declarationitemb",
  "modules.md#functionitema",
  "modules.md#functionitemb",
  "modules/namespacea.md",
  "classes/namespacea.somenestedclass.md",
  "classes/namespacea.somenestedclass.md#constructor",
  "modules/namespaceb.md",
  "enums/enumitema.md",
  "enums/enumitemb.md",
  "classes/classitema.md",
  "classes/classitema.md#constructor",
  "classes/classitemb.md",
  "classes/classitemb.md#constructor",
  "interfaces/interfaceitema.md",
  "interfaces/interfaceitema.md#prop",
  "interfaces/interfaceitemb.md",
  "interfaces/interfaceitemb.md#prop",
]
`;
